### ISSUE TITLE
generate with custom size fails

### ISSUE THREAD

**Issue by Kitton** [railsmechanic/nanoid#5]
Hello, looks like `Nanoid.generate(size)` doesn't generate an id with the desired length consistently.
You can easily verify it with the following script
```elixir
Enum.map(1..100, fn _x -> String.length(Nanoid.generate(30)) end) |> Enum.uniq()
[30, 29, 28]
```

**Comment by railsmechanic** [comment]
Many thanks for submitting. I‘ll have a look at it. 👍

**Comment by railsmechanic** [comment]
Found the problem. Will release v2.0.2 soon...

**Comment by railsmechanic** [comment]
**Fixed the issue and released nanoid v2.0.2**

The problem was that the`generator` function sometimes returns values which couldn't be mapped to valid characters, which leads to `nil` values, which are filtered out.
Now the `generate/1` function internally uses the same underlaying generation function as `generate/2`.

I've also added a test to validate that generated NanoIDs meet the requested size:
https://github.com/railsmechanic/nanoid/blob/a753ac69d4776989bb51e4bdac151a50b71220ed/test/nanoid/secure_test.exs#L42
